### PR TITLE
Update model share tutorial

### PR DESCRIPTION
**Pull request by stevhliu**
First draft of the updated model sharing tutorial. Main updates include:

- Suggest converting a model to other frameworks before uploading it to the Hub.
- Advocate for `push_to_hub` as best practice for uploading files to the Hub. Most notably, the `transformers-cli` method has been removed. Would love your thoughts on whether we should include using the web interface to share models so this step is more accessible a wider background of users!

### CODE REVIEW

**Comment by julien-c** [comment]
yes i would advocate to include using the web interface to share models

File: docs/source/model_sharing.mdx [hunk 1]
```diff
@@ -10,384 +10,122 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-
-Alternatively, you can use the `transformers-cli`. The next steps describe that process:
-
-Go to a terminal and run the following command. It should be in the virtual environment where you installed 🤗
-Transformers, since that command `transformers-cli` comes from the library.
-
-```bash
-transformers-cli login
+```py
+!huggingface-cli login
```
Comment (sgugger) [inline docs/source/model_sharing.mdx:54]: In a notebook we should advertise `notebook_login` which is more adapted.

**Review by sgugger — COMMENTED** [review]
Thanks for working on this, especially since the integration with the Hub has evolved a lot since the time of writing.

I think we should first show the most high-level tools: the `push_to_hub=True` argument when using the `Trainer` and the `PushToHubCallback` when using `Keras.fit` as it's the easiest to use.

Then only advertise the `push_to_hub` method on a model. Lastly it would be great to show how each repo behaves like a totally normal git repo if users want to edit the model card or add a new file.

File: docs/source/model_sharing.mdx [hunk 2]
```diff
@@ -10,384 +10,202 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-And that's all there is to it! It's also a very easy way to fix a mistake if one of the files online had a bug.
-
-To add a model for another backend, it's also super easy. Let's say you have fine-tuned a TensorFlow model and want to
-add the pytorch model files to your model repo, so that anyone in the community can use it. The following allows you to
-directly create a PyTorch version of your TensorFlow model:
+## Convert a model for all frameworks
 
-```python
-from transformers import AutoModel
+To ensure your model can be used by someone working with a different framework, we recommend you convert and upload your model with both PyTorch and TensorFlow checkpoints. While users are still able to load your model from a different framework if you skip this step, it will be slower because 🤗 Transformers will need to convert the checkpoint on-the-fly.
```
Comment (LysandreJik) [inline docs/source/model_sharing.mdx:74]: Maybe a quick mention of flax too, if the model is available in flax!

File: docs/source/model_sharing.mdx [hunk 3]
```diff
@@ -10,384 +10,202 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-
-
-### Add a model card
+Now when you navigate to the your Hugging Face profile, you should see your newly created model repository. Clicking on the **Files** tab will display all the files you've uploaded to the repository.
 
-To make sure everyone knows what your model can do, what its limitations, potential bias or ethical considerations are,
-please add a README.md model card to your model repo. You can just create it, or there's also a convenient button
-titled "Add a README.md" on your model page. A model card documentation can be found [here](https://huggingface.co/docs/hub/model-repos) (meta-suggestions are welcome). model card template (meta-suggestions
-are welcome).
+For more details on how to create and upload files to a repository, refer to the Hub documentation [here](https://huggingface.co/docs/hub/how-to-upstream).
```
Comment (LysandreJik) [inline docs/source/model_sharing.mdx:185]: Cool, important link

**Review by LysandreJik — APPROVED** [review]
Quite cool! Nice to mention the web interface, too.

**Pull request merged**